jib-core/src/main/java/com/google/cloud/tools/jib/configuration/LayerConfigurations.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.configuration;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/** Defines a list of {@link LayerConfiguration}s. */
+public class LayerConfigurations {
+
+  private final ImmutableList<LayerConfiguration> layerConfigurations;
+  private final ImmutableMap<String, LayerConfiguration> layerConfigurationMap;
```

Comment:
Seems strange to store both these?
Why not just `ImmutableMap`? it stores it in insertion order (like `LinkedHashMap`)?